### DISCUSSION TITLE
GitHub website's search bar feature request

### DISCUSSION THREAD

**Discussion by ayushmanchhabra** [community/community#23760] [New to GitHub]
<p>I noticed how the GitHub website’s search bar functions - specifically for users’ (person or organisation). I’d like to get some more information on how it works under the hood. Depending on that, I’d like to propose a feature request. As an example, let’s take my GitHub site’s repo header: <a href="https://github.com/ayushmxn?tab=repositories" rel="nofollow noopener">https://github.com/ayushmxn?tab=repositories</a></p>
<p>When I type anything in the search bar, it automatically searches for my query without waiting for the user to press enter. I’d like to know if it’s searching pre-loaded but currently hidden repository information on the website or if it is making an API call every time I make a change to the query. If the data searched is large, this can cause an unnecessary consumption of time and resources. If it is making an API call, then I would like to propose waiting for the user to click enter or/and adding a search button.</p>
<p>p.s.: Current this post is not categorized. Please let me know in which category this post could fit.</p>

**Comment by ghost** [comment, marked as answer]
<p>Hi <a class="mention" href="/ayushmxn">@ayushmxn</a>! Welcome to the GitHub Support Community! :partying_face:</p>
<p>Thanks for your considered feedback! We’re always working to improve GitHub and the GitHub Support Community, and we consider every suggestion we receive. Would you mind <a href="https://support.github.com/contact/feedback">submitting this through our official product feedback form</a> so that our product team can track your request?</p>
<p>Thanks so much for using GitHub and being a part of our community!</p>

**Comment by ayushmanchhabra** [comment]
<p>Done! :slight_smile:</p>
